### PR TITLE
Cv integration

### PR DESCRIPTION
**Pull request by spencerteetaert**
Addition of lane detection pipeline.

### CODE REVIEW

File: cv/cv/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    geometry_msgs
+    std_msgs
+)
+
+# Generate cmake config files via catkin_package
+catkin_package(
+  INCLUDE_DIRS
+  CATKIN_DEPENDS
+  message_runtime
+  )
```
Comment (1999michael) [inline cv/cv/CMakeLists.txt:30]: add newline

File: cv/cv/scripts/get_z_estimates.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+        avg_depth_vals = np.ma.mean(self.averages, axis=0)
+        avg_depth_vals = interpolate_nans(avg_depth_vals)
+
+        output = {}
+        for i, _y in enumerate(range(0, 180, 30)):
+            for j, _x in enumerate(range(0, 330, 30)):
+                for y in range(_y, _y+30):
+                    for x in range(_x, _x+30):
+                        output[str((x, y))] = avg_depth_vals[i, j]
```
Comment (1999michael) [inline cv/cv/scripts/get_z_estimates.py:40]: what are these numbers from?

Comment (1999michael) [reply cv/cv/scripts/get_z_estimates.py:40]: leave comment to add a filter if 30x30 grid thing proves to be insufficient?

File: cv/cv/scripts/zed_output_data.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    def run(self):
+        rospy.init_node('zed_data_output')
+        rospy.Subscriber("image", Image, self.save_image)
+        rospy.spin()
+
+    def save_image(self, data):
+        img = self.bridge.imgmsg_to_cv2(data, desired_encoding='passthrough')  
+
+        if self.delay == 0: 
+            cv2.imwrite('/home/spencer/Documents/ART/zed_images/4/' + str(self.counter) + '.png', img)
```
Comment (1999michael) [inline cv/cv/scripts/zed_output_data.py:31]: do I need to run this? or was this just for debugging. I don't have this folder

File: cv/cv/src/cv_utils/camera_projection.py [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+        detination image size: 180, 330
+        ratio: 0.5, 0.515625
+
+        TODO: Remove estimates 
+        '''
+
+        self.fx = (360 + 640) / 2
+        self.fy = (360 + 640) / 2
+        self.ox = 640 / 2
+        self.oy = 360 / 2
```
Comment (1999michael) [inline cv/cv/src/cv_utils/camera_projection.py:45]: mention a TO DO for getting camera intrinsics from ROS topic. Could be ok for now

**Review by 1999michael — APPROVED** [review]

**Review by kajananchinniah — APPROVED** [review]
Lgtm but please update to subscribe to camera info, etc (this isn't pretty but you promise it works)

Remind me to give it a proper review later

File: cv/cv/scripts/zed_output_data.py (hunk 3 above)
Comment (spencerteetaert) [reply cv/cv/scripts/zed_output_data.py:31]: Just for debugging. We use it to output data from zed for CV. So we collected pothole images with this

File: cv/cv/src/cv_utils/camera_projection.py (hunk 4 above)
Comment (spencerteetaert) [reply cv/cv/src/cv_utils/camera_projection.py:45]: Guys the TODO is literally already in the file lol

**Pull request merged**